torchbenchmark/score/compute_score_v1.py

```diff
@@ -0,0 +1,193 @@
+            domain_cnt = len(task_dict[type(task).__name__])
+            task_cnt = len(task_dict[type(task).__name__][task.name])
+            domain_weights[name] = (1.0 / category_cnt) * (1.0 / domain_cnt) * (1.0 / task_cnt)
+        # Setup config_weights
+        for name in name_dict:
+            (test, model_name, device, mode) = self._test_to_config(name)
+            config_weights[name] = config_dict[model_name][name] / sum(config_dict[model_name].values())
+        for name in name_dict:
+            total_weight = config_weights[name] * domain_weights[name]
+            sum_domain += total_weight
+        assert(abs(sum_domain - 1.0) < 1e-6), "The total task weights sum is not 1.0, please submit a bug report."
+        return (domain_weights, config_weights)
+ 
+    def _setup_benchmark_norms(self, ref_data):
+        """
+        Helper function which gets the normalization values per benchmark
+        by going through the reference data file.
+        """
+        if ref_data == TORCHBENCH_V1_REF_DATA:
+            with open(ref_data) as ref_file:
+                ref = yaml.full_load(ref_file)
+        else:
+            ref = self._get_norm_from_ref_data(ref_data)
+        return ref
+
+    def _get_norm_from_ref_data(self, ref_data):
+        """
+        This function iterates over the reference data (json object)
+        and calculates the normalization values based on the reference data.
+        It also sets up the domain weights of the score.
+        """
+        norm = {}
+        norm['benchmarks'] = {}
+        for b in ref_data['benchmarks']:
+            d = {}
+            d['norm'] = b['stats']['mean']
+            norm['benchmarks'][b['name']] = d
+        return norm
+
+    def _get_score(self, data, ref, ref_weights):
+        score = 0.0
+        (domain_weights, config_weights) = ref_weights
+        for name in data['benchmarks']:
+            norm = data['benchmarks'][name]['norm']
+            benchmark_score = domain_weights[name] * config_weights[name] * math.log(norm / ref['benchmarks'][name]['norm'])
+            score += benchmark_score
+        return math.exp(score)
+
+    def data_in_list(self, n, l):
+        for e in l:
+            if e not in n:
+                return False
+        return True
+
+    def _get_subscore(self, data, ref_norm, ref_weights, filters):
+        assert len(filters) == 2, error_msg
+        assert "cpu" in filters or "cuda" in filters, error_msg
+        assert "train" in filters or "eval" in filters, error_msg
+        score = 0.0
+        (domain_weights, _) = ref_weights
+        for name in filter(lambda x: self.data_in_list(x, filters), data['benchmarks']):
+            norm = data['benchmarks'][name]['norm']
+            benchmark_score = domain_weights[name] * math.log(norm / ref_norm['benchmarks'][name]['norm'])
+            score += benchmark_score
+        return math.exp(score)
+    
+    def compute_jit_speedup_score(self, data):
```

Comment:
Just checking terminology, is the JIT speedup a "subscore" or different score?

Reply:
It is a different score because it doesn't use the reference data. Instead, it only compares the jit speedup relative to the eager version of the test.